### PR TITLE
[#536] Linting

### LINKED ISSUE #536: Linting
Some variables do not have `!default`, meaning they cannot be overridden by users. We should add linting (*possibly the [stylelint-scss](https://github.com/kristerkari/stylelint-scss) stylelint plugin?), and fix all those that it finds

### PR DESCRIPTION
**Pull request by planktonic**
Fixes #536 

The following changes are contained in this pull request:

- Adds `stylelint-scss`, so we can lint the sass-specific syntax
- Fixes a few cases where variables were not declared with `!default`
- Disables the linting on the exceptions (all in `tools`, where we’re declaring a base value for a variable, then re-specifying it inside a conditional)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -81,6 +81,7 @@
     "style-loader": "^2.0.0",
     "stylelint": "^13.13.1",
     "stylelint-config-bitcrowd": "^4.0.0",
+    "stylelint-scss": "^3.20.1",
```
Comment (klappradla) [inline package.json:84]: Maybe mention in the README how to best run this (I guess it also adds an `npm` task)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
+# Unreleased
+
+## Fixed
+
+- All of the variables can now be overridden by users
+- Sizing of checkboxes & radios now works by setting the `$bitstyles-checkbox-size` and `$bitstyles-radio-size` variables respectively
```
Comment (klappradla) [inline CHANGELOG.md:6]: Is this "breaking" 👀 ?

File: scss/bitstyles/base/input-checkbox/_settings.scss [hunk 3]
```diff
@@ -6,7 +6,7 @@
 // Base styles ////////////////////////////////////////
 $border-radius: size.get('xxs') !default;
 $gap: size.get('xs') !default;
-$size: size.get('m') !default;
+$size: size.get('l') !default;
```
Comment (klappradla) [inline scss/bitstyles/base/input-checkbox/_settings.scss:9]: Was this a bug?

File: scss/bitstyles/tools/_color.scss [hunk 4]
```diff
@@ -15,10 +15,10 @@
 // @param {list} $percentages  A list of decmial values representing the percentages at which the mix-color will be mixed into the base-color
 
 @function generate-palette($base-color, $mix-color, $percentages) {
-  $color-map: ();
+  $color-map: (); // stylelint-disable-line scss/dollar-variable-default
```
Comment (klappradla) [inline scss/bitstyles/tools/_color.scss:18]: Hm, a bit weird that the linter does not get the function context here 🤔 

Does the fix from [this issue](https://github.com/kristerkari/stylelint-scss/issues/76) make a difference?

File: scss/bitstyles/tools/_directional-properties.scss [hunk 5]
```diff
@@ -2,6 +2,8 @@
 @use 'sass:list';
 @use 'sass:string';
 
+/* stylelint-disable scss/dollar-variable-default */
```
Comment (klappradla) [inline scss/bitstyles/tools/_directional-properties.scss:5]: See my earlier comment: I think/hope stylelint has an option for this already 😉

**Review by klappradla — CHANGES_REQUESTED** [review]
Yay for linting 💯 

I'd suggest to add a note about the linter in the contributors' section of the REAME. Plus: it probably makes sense to run it on CI.

The disabling of rules generates a lot of noise. Maybe you can quickly check if the outcomes of [this issue](https://github.com/kristerkari/stylelint-scss/issues/76) can make this easier for these cases?

File: package.json (hunk 1 above)
Comment (planktonic) [reply package.json:84]: There’s no new npm script needed — this is a plugin for stylelint that adds some sass-related/useful rules to the default set provided by stylelint. So when you run `yarn lint`, these rules are used too

File: CHANGELOG.md (hunk 2 above)
Comment (planktonic) [reply CHANGELOG.md:6]: Ahh, sorry this is something better factored out into its own PR — I‘ll do that!

File: scss/bitstyles/base/input-checkbox/_settings.scss (hunk 3 above)
Comment (planktonic) [reply scss/bitstyles/base/input-checkbox/_settings.scss]: Something I found along the way, which should be in its own PR (coming soon…)

File: package.json (hunk 1 above)
Comment (planktonic) [reply package.json:84]: …there was no section about linting there though, so I added one ✨

File: scss/bitstyles/tools/_color.scss (hunk 4 above)
Comment (planktonic) [reply scss/bitstyles/tools/_color.scss]: It did! I updated the linting rules, thanks 🎉

**Review by klappradla — APPROVED** [review]

**Pull request merged**